pandas/tseries/period.py

```diff
@@ -211,73 +214,53 @@ def _generate_range(cls, start, end, periods, freq, fields):
 
     @classmethod
     def _from_arraylike(cls, data, freq, tz):
```

Comment:
I would prob move this entire function into ``__new__``, then its all in 1 place (I don't think its called anywhere else)